Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp

```diff
@@ -341,8 +341,13 @@ void ThreadedCompositor::renderLayerTree()
 #endif
     }
 
-    if (viewportSize.isEmpty())
+    if (viewportSize.isEmpty()) {
+        // Make sure the compositing runloop goes back to idle state. Otherwise sub-sequent
+        // compositing requests will not be scheduled (because the runloop would be stuck in
+        // InProgress state).
```

Comment:
I think we need to understand how we end up here with an empty viewport. I think this early return should actually be an assert, since we should make sure we don't start a composition for an empty page.